app/src/test/java/com/nervousfish/nervousfish/util/CircularListTest.java

```diff
@@ -81,6 +81,7 @@ public void testEqualsHash() {
         list1.add(12);
         assertTrue(list3.equals(list4));
         assertTrue(list3.hashCode() == list4.hashCode());
+        assertFalse(list3.equals(null));
```

Comment:
Yeah, but there may be a bug in the equals method causing the result to be true when the input is null. We have to test this branch for full test coverage